### PR TITLE
TNG-730 Add Product-List Section (with tests)

### PR DESCRIPTION
**Pull request by vlapl**

### CODE REVIEW

File: src/components/ProductListItem/index.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  @Prop({ required: true }) title!: ProductListItemProps["title"];
+  @Prop({ required: true }) description!: ProductListItemProps["description"];
+  @Prop({ required: true }) price!: ProductListItemProps["price"];
+  @Prop() image: ProductListItemProps["image"];
+  @Prop() handleClick: ProductListItemProps["handleClick"];
+
+  render() {
+    return (
+      <div class={`ProductListItem`}>
+        <img src={this.image?.src} />
```
Comment (tgolbs) [inline src/components/ProductListItem/index.tsx:22]: Please use the `<Image />` Component provided by fsxa-ui

File: src/components/ProductListItem/style.css [hunk 2]
```diff
@@ -0,0 +1,80 @@
+.ProductListItem {
```
Comment (tgolbs) [inline src/components/ProductListItem/style.css:1]: You can remove most of the classes by using the utility classes provided by tailwindcss.

Comment (tgolbs) [reply src/components/ProductListItem/style.css:1]: We can do this together if you want to.

File: src/components/ProductListItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import BaseComponent from "@/components/BaseComponent";
+import { Component, Prop } from "vue-property-decorator";
+import Headline from "@/components/Headline";
+import RichText from "@/components/RichText";
+
+import "./style.css";
+import { ProductListItemProps } from "@/types/components";
+
+@Component({
+  name: "FSXAProductListItem",
```
Comment (tgolbs) [inline src/components/ProductListItem/index.tsx:10]: Please change the "displayName"  to `ProductListItem`

File: src/components/Sections/ProductListSection/index.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        title={item.title}
+        description={item.description}
+        price={item.price}
+        image={item.image}
+      />
+    );
+  }
+
+  render() {
+    console.log("items: ", this.items);
```
Comment (tgolbs) [inline src/components/Sections/ProductListSection/index.tsx:29]: Please remove the console.log

File: src/components/Sections/ProductListSection/index.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import ProductListItem from "@/components/ProductListItem";
+
+@Component({
+  name: "ProductListSection",
+})
+class ProductListSection extends BaseComponent<ProductListSectionProps> {
+  @Prop({ required: true }) headline!: ProductListSectionProps["headline"];
+  @Prop({ required: true }) items!: ProductListSectionProps["items"];
+
+  renderItem(item: ProductListItemProps) {
```
Comment (tgolbs) [inline src/components/Sections/ProductListSection/index.tsx:17]: You could render the item inline in line 36 as well.

File: src/documentation/examples/ProductListSection.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    image: {
+      src:
+        "https://enterprisedev.e-spirit.cloud/smartlivingglobal/media/images/product-images/WLAN-powerplug_product_teaser.jpg",
+    },
+  },
+];
+
+@Component
+export default class App extends Vue {
+  currentItem: any = null;
```
Comment (tgolbs) [inline src/documentation/examples/ProductListSection.tsx:50]: Please remove this local state.

File: src/documentation/examples/ProductListSection.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+  render() {
+    return (
+      <div>
+        <ProductListSection
+          headline={headline}
+          items={items}
+        ></ProductListSection>
```
Comment (tgolbs) [inline src/documentation/examples/ProductListSection.tsx:58]: If you do not pass content as a child you can omit the closing `</ProductListSection>` and do this instead:

```
<ProductListSection
          headline={headline}
          items={items}
/>
```

File: src/types/components.d.ts [hunk 8]
```diff
@@ -318,6 +318,15 @@ export interface NewsTeaserItemProps {
 }
 export class NewsTeaserItem extends Component<NewsTeaserItemProps> {}
 
+export interface ProductListItemProps {
+  title: string;
+  description: string;
+  price?: number;
```
Comment (tgolbs) [inline src/types/components.d.ts:324]: You declared the price as required in your prop-decorator. Please synchronize the type with the property definition.

File: src/types/sections.d.ts [hunk 9]
```diff
@@ -1,6 +1,7 @@
 import { Component } from "vue-tsx-support";
 import { ImageRef } from "./utils";
 import { NewsTeaserItemProps } from "./components";
+import { ProductListItemProps } from "./components";
```
Comment (tgolbs) [inline src/types/sections.d.ts:4]: Please merge the imports into one single statement.

**Review by tgolbs — COMMENTED** [review]
This looks pretty great for a first draft. I've already add comments to some lines for you to improve.

File: src/components/ProductListItem/index.tsx (hunk 3 above)
Comment (vlapl) [reply src/components/ProductListItem/index.tsx:10]: done

**Pull request merged**